main.lua

```diff
@@ -1366,3 +1350,71 @@ function userdataIterator(data)
         if success then return idx, item end
     end
 end
+
+local LSPCmds = {
+    ["start"] = startServer,
+    ["stop"] = stopServers,
+    ["showlog"] = showLog,
+    ["sync-document"] = function (bp) syncFullDocument(bp.Buf) end,
+    ["hover"] = hoverAction,
+    ["format"] = formatAction,
+    ["autocomplete"] = completionAction,
+    ["goto-definition"] = gotoAction("definition"),
+    ["goto-declaration"] = gotoAction("declaration"),
+    ["goto-typedefinition"] = gotoAction("typeDefinition"),
+    ["goto-implementation"] = gotoAction("implementation"),
+    ["find-references"] = findReferencesAction,
+    ["document-symbols"] = documentSymbolsAction,
+    ["diagnostic-info"] = openDiagnosticBufferAction,
+}
+
+local function lspCompleter(buf)
+    local opts = {}
+
+    --Do NOT autocomplete after first argument
+    local args = go_strings.Split(buf:Line(0), " ")
+    if #args > 2 then return nil, nil end
+
+    for k, _ in pairs(LSPCmds) do table.insert(opts, k) end
+    table.sort(opts)
+
+    --NOTE assumes no unicode in plugin's options
```

Comment:
Could you clarify what this means? In what situation would unicode cause problems?